internal/config/validation.go

```diff
@@ -96,6 +97,53 @@ func DiscardNativeOnly(c ClusterResources) error {
 	return nil
 }
 
+// validateConfig is meant to validate all the inter-dependencies of a parsed configuration.
+// In this case, we ensure that bfd echo is not enabled on a pool.
+func validateConfig(cfg *Config) error {
+outer:
+	for _, p := range cfg.Pools.ByName {
+		containsV6 := false
+		for _, cidr := range p.CIDR {
+			if ipfamily.ForCIDR(cidr) == ipfamily.IPv6 {
+				containsV6 = true
+				break
+			}
+			if !containsV6 { // we only care about v6 advertisements
```

Comment:
not sure I understand, if the first cidr is ipv4 and the next is ipv6 we continue to the next pool?
I thought any pool that has an ipv6 cidr should be processed - meaning removing the outer loop, having this if outside of the cidrs loop? can you leave a comment there what exactly is a config we're trying to avoid?

Reply:
no this is wrong, let me fix it